packages/grid/x-data-grid/src/models/gridStateCommunity.ts

```diff
@@ -45,7 +45,7 @@ export interface GridStateCommunity {
  * The initial state of `DataGrid`.
  */
 export interface GridInitialStateCommunity {
-  pagination?: GridPaginationInitialState;
+  paginationModel?: GridPaginationInitialState;
```

Comment:
Named it `paginationModel` to be aligned with `props.paginationModel`.

If we still want to name it `pagination`, then maybe we should rename the `pagination` boolean prop to be `disablePagination` and use `pagination` everywhere instead of `paginationModel`. But it seems to cause more problems than it solves.

Let me know if `paginationModel` looks off.

Reply:
> I would prefer the sub-state to be named like the plugin.
`state.detailPanel` but `props.detailPanelExpandedRowIds`

That's what I wanted to do too, but then the idea of calling it in `initialState` as `pagination` and in props as `paginationModel` (example no. 1 below) didn't look great to me. 😄

```tsx
// 1. With `state.pagination` and `initialState.pagination`
<DataGrid
  initialState={{ pagination: { page: 0, pageSize: 10 } }}
  paginationModel={{ page: 0, pageSize: 10 }}
/>

// 2. With `state.paginationModel` and `initialState.paginationModel`
<DataGrid
  initialState={{ paginationModel: { page: 0, pageSize: 10 } }}
  paginationModel={{ page: 0, pageSize: 10 }}
/>

// 3. With `state.pagination` and `initialState.paginationModel`
<DataGrid
  initialState={{ paginationModel: { page: 0, pageSize: 10 } }}
  paginationModel={{ page: 0, pageSize: 10 }}
/>
```

I wanted to go with no. 2 as it logically aligned better to me as an end user with single level state but we can go with no. 1 too if it looks ok to team?

Reply:
For me `3.` is the worse`, when I introduced `initialState` I built it to be a sub-set of `state`.

Reply:
It would be great if we make it consistent with other initialState properties, like `sorting` and `filter`:

```tsx
<DataGrid
  initialState={{
    sorting: {
      sortModel: [{ field: 'plan', sort: 'asc' }]
    },
    filter: {
      filterModel: { items: [{ field: 'title', operator: 'equals', value: 'Titanic' }] }
    },
    pagination: {
      paginationModel: { page: 0, pageSize: 10 }
    }
  }}
  paginationModel={{ page: 0, pageSize: 10 }}
/>
```

What do you think?

Reply:
It was suggested to make pagination state single level in [this](https://github.com/mui/mui-x/pull/7147#discussion_r1062371217) thread.

Do you mean just keeping initialState interface like this while keeping state single level?

```tsx
interface GridPaginationModel {
  page: number;
  pageSize: number;
}

interface GridPaginationState extends GridPaginationModel {}

interface GridPaginationInitialState {
  paginationModel?: Partial<GridPaginationModel>;
}

const initialState = {
  pagination: {
    paginationModel: { page: 0, pageSize: 10 }
  }
}
```

Reply:
As discussed with the team, we'll be going with Andrew's [suggestion](https://github.com/mui/mui-x/pull/7147#discussion_r1069682438) on this topic